File: 0982-minimum-increment-to-make-array-unique/0982-minimum-increment-to-make-array-unique.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int ans = 0, need = 0;
        for(int n : nums) {
            ans += max(need - n, 0);
            need = max(n, need) + 1;
        }

        return ans;
    }
};

File: 0982-minimum-increment-to-make-array-unique/0982-minimum-increment-to-make-array-unique.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minIncrementForUnique = function(nums) {
    nums.sort((a, b) => a - b);
    let ans = 0, need = 0;
    for(let n of nums) {
        ans += Math.max(need - n, 0);
        need = Math.max(n, need) + 1;
    }

    return ans;
};